### PR TITLE
front:  Migrate InfraSelectorModalBodyEdition from js to ts 

### LINKED ISSUE #3843: Front: Wrong import endpoint for the infra
### What happened?

The front use the API endpoint to import a infra.

### What did you expect to happen?

The front should use the editoast endpoint to import an infra.

### How can we reproduce it (as minimally and precisely as possible)?

1. Try to import an infra

### What operating system, browser and environment are you using?

- **Browser**: Chrome v110
- **OS**: Windows 10
- **Env**: Dev Dgex

### OSRD version (top right corner `Account` button > `Informations`)

5cfdb304

### LINKED ISSUE #3947: Migrate InfraSelectorModalBodyEdition from js to ts
### Description and goal

Refacto InfraSelectorModalBodyEdition to migrate from js to ts

### Acceptance criteria

- [x] InfraSelectorModalBodyEdition.jsx is migrated to InfraSelectorModalBodyEdition.tsx

### PR DESCRIPTION
**Pull request by Uriel-Sautron**
Migrate InfraSelectorModalBodyEdition from js to ts and use RTK
close #3947 
close #3843

### CODE REVIEW

File: front/src/common/InfraSelector/InfraSelectorModalBodyEdition.tsx [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
+  infrasList: Infra[];
+  setFilter: React.Dispatch<React.SetStateAction<string>>;
+  filter: string;
+  getInfrasList: () => void;
+};
+
+const InfraSelectorModalBodyEdition = ({
+  infrasList,
+  setFilter,
+  filter,
```
Comment (Math-R) [inline front/src/common/InfraSelector/InfraSelectorModalBodyEdition.tsx]: I think you forget to redefine the default value to => ''

File: front/src/common/InfraSelector/InfraSelectorModal.tsx [hunk 2]
```diff
@@ -4,51 +4,52 @@ import ModalHeaderSNCF from 'common/BootstrapSNCF/ModalSNCF/ModalHeaderSNCF';
...
-  function filterInfras(filteredInfrasListLocal: InfraListType) {
+  function filterInfras(filteredInfrasListLocal: Infra[]) {
     if (debouncedFilter && debouncedFilter !== '') {
-      filteredInfrasListLocal = infrasList.filter((infra: InfraType) =>
+      filteredInfrasListLocal = infrasList.filter((infra) =>
         infra.name.toLowerCase().includes(debouncedFilter.toLowerCase())
       );
     }
-    filteredInfrasListLocal.sort((a, b) => a.name.localeCompare(b.name));
+    filteredInfrasListLocal.slice().sort((a, b) => a.name.localeCompare(b.name));
```
Comment (Math-R) [inline front/src/common/InfraSelector/InfraSelectorModal.tsx]: Slice not use full

**Review by Math-R — CHANGES_REQUESTED** [review]

File: front/src/common/InfraSelector/InfraSelectorModalBodyEdition.tsx (hunk 1 above)
Comment (Uriel-Sautron) [reply front/src/common/InfraSelector/InfraSelectorModalBodyEdition.tsx]: Done !!!

File: front/src/common/InfraSelector/InfraSelectorModal.tsx (hunk 2 above)
Comment (Uriel-Sautron) [reply front/src/common/InfraSelector/InfraSelectorModal.tsx]: Done, thank you for review !!

**Review by Math-R — APPROVED** [review]

**Pull request merged**